### PR TITLE
Create special RequireThis configs with custom set of project

### PR DESCRIPTION
**Pull request by relentless-pursuit**
Resolves: #90

### CODE REVIEW

File: RequireThis/all-examples-in-one/list-of-projects.properties [hunk 1]
```diff
@@ -37,3 +37,5 @@ protonpack|git|https://github.com/poetix/protonpack|protonpack-1.7||
 jOOL|git|https://github.com/jOOQ/jOOL|version-0.9.7||
 RxJava|git|https://github.com/ReactiveX/RxJava|v1.0.9||
 Vavr|git|https://github.com/vavr-io/vavr|v0.9.0||
+
+checkstyle|git|https://github.com/checkstyle/checkstyle.git|master|**/.ci-temp/**/*,**/resources-noncompilable/**/asttreestringprinter/**/*,**/resources-noncompilable/**/filefilters/**/*,**/resources-noncompilable/**/main/**/*,**/resources-noncompilable/**/suppressionsstringprinter/**/*,**/resources-noncompilable/**/gui/**/*,**/resources-noncompilable/**/javadocpropertiesgenerator/**/*,src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/javaparser/InputJavaParser.java,**/InputAllEscapedUnicodeCharacters.java,**/resources-noncompilable/**/javaparser/InputJavaParser.java,**/resources-noncompilable/**/checks/imports/unusedimports/InputUnusedImportsSingleWordPackage.java,**/resources-noncompilable/**/grammar/java19/*,**/resources-noncompilable/**/treewalker/**/*
```
Comment (romani) [inline RequireThis/all-examples-in-one/list-of-projects.properties:41]: ??
https://github.com/checkstyle/contribution/blob/d85c3ddc6ae531dc372b83509be88a1b15b86ac3/checkstyle-tester/projects-to-test-on-for-github-action.properties#L40-L41

**Review by romani — CHANGES_REQUESTED** [review]
Item

File: RequireThis/all-examples-in-one/list-of-projects.properties (hunk 1 above)
Comment (relentless-pursuit) [reply RequireThis/all-examples-in-one/list-of-projects.properties]: My bad has pushed again. Please let me know if any issue persists.

File: extractor/src/main/resources/all-projects.properties [hunk 2]
```diff
@@ -39,4 +39,19 @@ RxJava|git|https://github.com/ReactiveX/RxJava|v1.0.9||
...
+
+guava|git|https://github.com/google/guava|v28.2||
+
+# custom javadoc tags
+nbia-dcm4che-tools|git|https://github.com/thprakash/nbia-dcm4che-tools|c3591e6f0f84827586db25abded6708e5386ef1a||
+
+# RequireThis usage
+spring-integration|git|https://github.com/spring-projects/spring-integration|main|
+
+javadoc/tool/T4994049/FileWithTabs.java,**/test/langtools/tools/javac/annotations/neg/pkg/package-info.java|
```
Comment (romani) [inline extractor/src/main/resources/all-projects.properties:57]: What is this?

File: extractor/src/main/resources/all-projects.properties [hunk 3]
```diff
@@ -39,4 +39,19 @@ RxJava|git|https://github.com/ReactiveX/RxJava|v1.0.9||
 Vavr|git|https://github.com/vavr-io/vavr|v0.9.0||
 
 # japan language
-WxJava|git|https://github.com/Wechat-Group/WxJava.git|develop|
\ No newline at end of file
+WxJava|git|https://github.com/Wechat-Group/WxJava.git|develop|
+
+# Few projects that delivers set of unusual Java constructions that shall be correctly handled by AST visitor
+checkstyle|git|https://github.com/checkstyle/checkstyle.git|master|
+sevntu-checkstyle|git|https://github.com/sevntu-checkstyle/sevntu.checkstyle|master||
```
Comment (romani) [inline extractor/src/main/resources/all-projects.properties:46]: We already have this above, why we need it again?

**Review by romani — CHANGES_REQUESTED** [review]
Please add config to yml file to generate extra project to RequireThis

**Comment by romani** [comment]
Please review your PR yourself, it helps to avoid unexpected.

File: extractor/src/main/resources/projects-for-example.yml [hunk 4]
```diff
@@ -4,6 +4,44 @@ SummaryJavadoc:
...
+  Example6:
+    projects:
+      - default
+      - spring-integration
+  all-examples-in-one:
+    projects:
+      - default
+      - spring-integration
+
+# More checks and examples can be added as needed
```
Comment (romani) [inline extractor/src/main/resources/projects-for-example.yml:37]: Remove this

File: extractor/src/main/resources/projects-for-example.yml [hunk 5]
```diff
@@ -4,6 +4,44 @@ SummaryJavadoc:
       - WxJava
       - openjdk17
 
+RequireThis:
+  Example1:
+    projects:
+      - default
```
Comment (romani) [inline extractor/src/main/resources/projects-for-example.yml:10]: What is default?
Better to reference there all projects

**Review by romani — CHANGES_REQUESTED** [review]
Item

File: extractor/src/main/resources/projects-for-example.yml (hunk 5 above)
Comment (relentless-pursuit) [reply extractor/src/main/resources/projects-for-example.yml]: Left default, now referencing all projects. Done.

File: extractor/src/main/resources/all-projects.properties [hunk 6]
```diff
@@ -39,4 +39,7 @@ RxJava|git|https://github.com/ReactiveX/RxJava|v1.0.9||
 Vavr|git|https://github.com/vavr-io/vavr|v0.9.0||
 
 # japan language
-WxJava|git|https://github.com/Wechat-Group/WxJava.git|develop|
\ No newline at end of file
+WxJava|git|https://github.com/Wechat-Group/WxJava.git|develop|
+
+# RequireThis usage
+spring-integration|git|https://github.com/spring-projects/spring-integration|main|
```
[the remark this answers is not in the document]
Comment (relentless-pursuit) [reply extractor/src/main/resources/all-projects.properties]: done

**Review by romani — APPROVED** [review]

**Pull request merged**